src/components/ShelterListItem/ShelterListItem.tsx

```diff
@@ -4,121 +4,144 @@ import { useNavigate } from 'react-router-dom';
+      .slice(0, 10);
+  }, [data.shelterSupplies]);
 
-  const donationsTags = useMemo(
-    () => {
-      return data.shelterSupplies?.filter((s) => !getCategoriesToFilterVolunteers().some(c => c.includes(s.supply?.supplyCategory?.name.toLowerCase())) && s.priority === SupplyPriority.Remaining).reverse()
-    },
-    [data.shelterSupplies]
-  )
+  const volunteerTags = useMemo(() => {
+    return data.shelterSupplies
+      ?.filter((s) =>
+        getCategoriesToFilterVolunteers().some((c) =>
+          c.includes(s.supply?.supplyCategory?.name.toLowerCase())
+        )
+      )
+      .sort((a, b) => b.priority - a.priority);
+  }, [data.shelterSupplies]);
 
+  const donationsTags = useMemo(() => {
+    return data.shelterSupplies
+      ?.filter(
+        (s) =>
+          !getCategoriesToFilterVolunteers().some((c) =>
+            c.includes(s.supply?.supplyCategory?.name.toLowerCase())
+          ) && s.priority === SupplyPriority.Remaining
+      )
+      .reverse();
+  }, [data.shelterSupplies]);
 
   return (
     <div
-      className="flex flex-col p-4 w-full border-2 border-border rounded-md gap-1 relative"
+      className="relative flex flex-col w-full gap-1 p-4 border-2 rounded-md border-border"
       onClick={() => navigate(`/abrigo/${data.id}`)}
     >
       <Button size="sm" variant="ghost" className="absolute top-4 right-4">
-        <ChevronRight className="h-5 w-5" />
+        <ChevronRight className="w-5 h-5" />
       </Button>
       <div className="flex items-center gap-1">
-        <h3 className="font-semibold text-lg  hover:cursor-pointer hover:text-slate-500">
+        <h3 className="text-lg font-semibold hover:cursor-pointer hover:text-slate-500">
           {data.name}
         </h3>
         {data.verified && <VerifiedBadge />}
       </div>
-      <h6 className={cn('font-semibold text-md', availabilityClassName)}>
-        {availability}
-      </h6>
-      <h6 className="text-muted-foreground text-sm md:text-lg font-medium">
-        {data.address}
-      </h6>
+      <h6 className={cn('font-semibold text-md', availabilityClassName)}>{availability}</h6>
+      <h6 className="text-sm font-medium text-muted-foreground md:text-lg">{data.address}</h6>
       {data.shelterSupplies.length > 0 && (
         <>
           <div className="flex flex-col gap-3">
             <Separator className="mt-2" />
-            <p className="text-muted-foreground text-sm md:text-lg font-medium">
+            <p className="text-sm font-medium text-muted-foreground md:text-lg">
               Necessita voluntários:
             </p>
-            <div className="flex gap-2 flex-wrap">
-              {volunteerTags.length == 0 ?
-                <p> Não informado. <i> (Pode ser adicionado ao clicar no abrigo) </i></p>
-                :
+            <div className="flex flex-wrap gap-2" onClick={(e) => e.stopPropagation()}>
+              {Array.isArray(volunteerTags) && volunteerTags.length > 0 ? (
```

Comment:
Aqui poderia ser apenas como abaixo não?
```suggestion
              {volunteerTags?.length ? (
```